ci-operator/step-registry/cucushift/installer/check/aws/custom-ami/cucushift-installer-check-aws-custom-ami-commands.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+REGION="${LEASED_RESOURCE}"
+INFRA_ID=$(jq -r '.infraID' ${SHARED_DIR}/metadata.json)
+export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+
+if [ -f "${SHARED_DIR}/kubeconfig" ] ; then
+  export KUBECONFIG=${SHARED_DIR}/kubeconfig
+else
+  echo "No KUBECONFIG found, exit now"
+  exit 1
+fi
+
+CONFIG=${SHARED_DIR}/install-config.yaml
+if [ ! -f "${CONFIG}" ] ; then
+  echo "No install-config.yaml found, exit now"
+  exit 1
+fi
+
+
+case "${OCP_ARCH}" in
+amd64)
+    ARCH="x86_64"
+    ;;
+arm64)
+    ARCH="aarch64"
+    ;;
+*)
+    echo "${OCP_ARCH} is not supported, exit now"
+    exit 1
+esac
+
+# default_ami=$(openshift-install coreos print-stream-json | jq -r --arg a $ARCH --arg r $REGION '.architectures[$a].images.aws.regions[$r].image')
+# echo "AMI (default, \"openshift-install coreos print-stream-json\"): ${default_ami}"
+
+control_plane_ami=$(aws --region $REGION ec2 describe-instances --filters "Name=tag:kubernetes.io/cluster/${INFRA_ID},Values=owned" "Name=tag:Name,Values=*master*" --output json | jq -r '.Reservations[].Instances[].ImageId' | sort | uniq)
+compute_ami=$(aws --region $REGION ec2 describe-instances --filters "Name=tag:kubernetes.io/cluster/${INFRA_ID},Values=owned" "Name=tag:Name,Values=*worker*" --output json | jq -r '.Reservations[].Instances[].ImageId' | sort | uniq)
+echo "AMI used by cluster: controlPlane: [${control_plane_ami}], compute: [${compute_ami}]"
+
+ret=0
+
+ic_platform_ami=$(yq-go r "${CONFIG}" 'platform.aws.amiID')
+ic_control_plane_ami=$(yq-go r "${CONFIG}" 'controlPlane.platform.aws.amiID')
+ic_compute_ami=$(yq-go r "${CONFIG}" 'compute[0].platform.aws.amiID')
+echo "AMI in install-config: platform: [${ic_platform_ami}], controlPlane: [${ic_control_plane_ami}], compute: [${ic_compute_ami}]"
+
+
+if [[ "${ic_platform_ami}" != "" ]] && [[ "${ic_platform_ami}" != "null" ]]; then
```

Comment:
Can we apply the same logic like https://github.com/openshift/release/pull/52689#discussion_r1668114731 here ?